### PR TITLE
Migrate paul_revere gem functionality into src

### PR DESCRIPTION
**Pull request by colby-swandale**
This is part of the work i'm doing to slowly move rubygems.org over to Rails 5.

The gem [thoughtbot/paul_revere](https://github.com/thoughtbot/paul_revere) appears to no longer be maintained by the authors.

This PR moves the the main functionality in the `paul_revere` gem into src and removes the gem from the `Gemfile`

Thanks 🎉

### CODE REVIEW

**Comment by sonalkr132** [comment]
> The command "script/brakeman" exited with 1.

It fails for bundler 1.14.0 and above. Perhaps brakeman 3.6 is not compatible with bundler 1.14?

File: app/views/layouts/_announcement.html.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<% if announcement_visible?(current_announcement) %>
+  <div id="announcement">
+    <span><%= current_announcement.body.html_safe %></span>
+    <span class="hide">
+      <%= link_to "hide", "#", onclick: "hideAnnouncement('#{current_announcement.token}')" %>
```
Comment (sonalkr132) [inline app/views/layouts/_announcement.html.erb]: Please don't use inline javascript, you would end up violating [our CSP](https://github.com/rubygems/rubygems.org/blob/master/app/controllers/application_controller.rb#L15). Perhaps you can add it to `announcement,js`.

File: app/assets/javascripts/announcements.js [hunk 2]
```diff
@@ -1,29 +1,36 @@
...
-    while (c.charAt(0) == ' ') c = c.substring(1, c.length);
-    if (c.indexOf(nameEQ) == 0) return c.substring(nameEQ.length, c.length);
+  hideAnnouncement.on('click', function() {
+    announcement_key = announcement.data("announcement-token");
+    createCookie(announcement_key, "hidden", 365);
+    $("#announcement").slideUp();
+  });
+
+  // Utility scripts from http://www.quirksmode.org/js/cookies.html
+  function createCookie(name, value, days) {
```
Comment (jvanbaarsen) [inline app/assets/javascripts/announcements.js]: Would it make sense to move this utility scripts to its own file called: `cookies.js`?

File: app/models/announcement.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Announcement < ActiveRecord::Base
+  def token
+    "#{self.class.name.downcase}_#{created_at.iso8601}"
```
Comment (jvanbaarsen) [inline app/models/announcement.rb]: What is the reason we are parsing the `created_at` to a specific iso standard? Did the default rails `created_at` give issues?

File: app/assets/javascripts/announcements.js [hunk 4]
```diff
@@ -1,29 +1,36 @@
...
-function createCookie(name, value, days) {
-  if (days) {
-    var date = new Date();
-    date.setTime(date.getTime() + (days * 24 * 60 * 60 * 1000));
-    var expires = "; expires=" + date.toGMTString();
-  } else var expires = "";
-  document.cookie = name + "=" + value + expires + "; path=/";
-}
+  var announcement = $('#announcement');
+  var hideAnnouncement = $('#announcement .hide > a');
```
Comment (jvanbaarsen) [inline app/assets/javascripts/announcements.js]: What do you think about using `data-attributes` for this? That way we are not relying on the HTML. For example we could add `data-behaviour="hide-announcement"` to the link.

**Comment by dwradcliffe** [comment]
@colby-swandale I guess we waited long enough that we don't need this anymore. Thanks for your work here anyway!

**Comment by simi** [comment]
@dwradcliffe also issue mentioned in first post was already solved in upstream and in Rails 5 upgrade in this repo.

**Pull request closed without merging**